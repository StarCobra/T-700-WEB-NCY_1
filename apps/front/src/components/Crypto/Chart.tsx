import React from "react";
import { Box } from "@mui/material";
import ReactApexChart from "react-apexcharts";

export default function Chart(props: any) {
  const { resource, title } = props;

  // Graphic's options
  const options = {
    chart: {
      type: "candlestick",
      height: 350,
      background: "#b7b7b7", // Couleur de fond blanche
    },
    title: {
      text: title,
      align: "left",
    },
    xaxis: {
      type: "datetime",
    },
    yaxis: {
      tooltip: {
        enabled: true,
        style: {
          fontSize: "14px",
          color: "#000000",
        },
      },
    },
    legend: {
      show: false,
    },
  };

  const data = resource.map((item: any, index: number) => ({
    x: new Date(1538785000000 + index * 86400000),
    y: [
      item?.change_price_day, // Open price
      item?.current_price, // Close price
      item?.highest_price_day, // Highest price
      item?.lowest_price_day, // Lowest price
    ],
  }));

  return (
    <Box width="600px" style={{ color: "#000000" }}>
      <ReactApexChart
        // eslint-disable-next-line
        // @ts-ignore
        options={options}
        series={[{ data }]}
        type="candlestick"
      />
    </Box>
  );
}
