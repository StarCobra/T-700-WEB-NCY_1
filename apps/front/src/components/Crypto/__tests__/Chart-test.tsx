import React from "react";
import { act, create } from "react-test-renderer";
import Chart from "../Chart";

describe("Components > Crypto > Chart", () => {
  const tests = [
    {
      description: "generic",
      props: {},
    },
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;
      await act(async () => {
        root = create(<Chart {...test.props} />);
      });
      expect(root).toMatchSnapshot();
    });
  });
});
