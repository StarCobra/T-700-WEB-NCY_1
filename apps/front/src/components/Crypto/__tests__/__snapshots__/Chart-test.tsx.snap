// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Crypto > Chart should render the component generic 1`] = `
<div
  className="MuiBox-root css-0"
>
  <p>
    theCryptoChart
  </p>
  ;
</div>
`;
